test/cassandrax/keyspace_test.exs

```diff
@@ -0,0 +1,377 @@
+    @describetag seeds: [@zero]
+
+    test "insert valid data" do
+      assert schema_equal(TestKeyspace.insert(@one), {:ok, @one})
+    end
+
+    test "insert! valid data" do
+      assert schema_equal(TestKeyspace.insert!(@one), @one)
+    end
+
+    test "insert invalid data" do
+      assert {:error, :invalid_data} == TestKeyspace.insert(@invalid_one)
+    end
+
+    test "insert! invalid data" do
+      assert_raise(Cassandrax.InvalidDataError, fn ->
+        TestKeyspace.insert!(@invalid_one)
+      end)
+    end
+
+    test "update valid data" do
+      changeset = Changeset.change(@zero, value: "new zero")
+      expectation = %{@zero | value: "new zero"}
+      assert schema_equal(TestKeyspace.update(changeset), {:ok, expectation})
+      assert schema_equal(TestKeyspace.get(TestData, id: "0"), expectation)
+    end
+
+    test "update! valid data" do
+      changeset = Changeset.change(@zero, value: "new zero")
+      expectation = %{@zero | value: "new zero"}
+      assert schema_equal(TestKeyspace.update!(changeset), expectation)
+      assert schema_equal(TestKeyspace.get(TestData, id: "0"), expectation)
+    end
+
+    test "update invalid data" do
+      changeset = Changeset.change(@zero, value: 1)
+      assert  {:error, :invalid_data} == TestKeyspace.update(changeset)
+    end
+
+    test "update! invalid data" do
+      changeset = Changeset.change(@zero, value: 1)
+      assert_raise(Cassandrax.InvalidDataError, fn ->
+        TestKeyspace.update!(changeset)
+      end)
+    end
+
+    test "delete valid data" do
+      assert schema_equal(TestKeyspace.delete(@zero), {:ok, @zero})
+      assert TestKeyspace.get(TestData, id: "0") == nil
+    end
+
+    test "delete! valid data" do
+      assert schema_equal(TestKeyspace.delete!(@zero), @zero)
+      assert TestKeyspace.get(TestData, id: "0") == nil
+    end
+
+    test "delete invalid id" do
+      assert {:error, :invalid_data} == TestKeyspace.delete(@invalid_id)
+    end
+
+    test "delete! invalid id" do
+      assert_raise(Cassandrax.InvalidDataError, fn ->
+        TestKeyspace.delete!(@invalid_id)
+      end)
+    end
+  end
+
+  describe "queryables" do
+    @describetag seeds: [@zero, @one]
+
+    test "get" do
+      assert TestKeyspace.get(TestData, id: "0") == @zero
+      assert TestKeyspace.get(TestData, id: "3") == nil
+    end
+
+    test "all" do
+      assert list_sets_equal?(TestKeyspace.all(TestData), [@zero, @one])
+    end
+  end
+
+  describe "one" do
+    test "fails on table with no entries" do
+      assert TestKeyspace.one(TestData) == nil
+    end
+
+    test "from table with one entry" do
+      TestKeyspace.insert!(@zero)
+
+      assert TestKeyspace.one(TestData) == @zero
+    end
+
+    test "from table with multiple entries" do
+      TestKeyspace.insert!(@zero)
+      TestKeyspace.insert!(@one)
+
+      assert TestKeyspace.one(where(TestData, id: "0")) == @zero
+      assert TestKeyspace.one(where(TestData, id: "1")) == @one
```

Comment:
things like this you can use the pipe syntax like so:
```
TestData |> where(id: "0") |> TestKeyspace.one()
```

Reply:
ok, I can use pipe here.